
<launch>

   <!-- ROBOT MAPPING VERSION: use this with ROS bag demo_mapping.bag -->
   <!-- WARNING : Database is automatically deleted on each startup -->
   <!--           See "delete_db_on_start" option below... -->

  <param name="use_sim_time" type="bool" value="True"/>

  <group ns="rtabmap">
    <!-- SLAM (robot side) -->
    <!-- args: "delete_db_on_start" and "udebug" -->
    <node name="rtabmap" pkg="rtabmap" type="rtabmap" output="screen" args="--delete_db_on_start">
	  <param name="frame_id" type="string" value="base_footprint"/>
	
	  <param name="subscribe_depth" type="bool" value="true"/>
	  <param name="subscribe_laserScan" type="bool" value="true"/>
	
	  <remap from="odom" to="/az3/base_controller/odom"/>
	  <remap from="scan" to="/jn0/base_scan"/>

	  <remap from="rgb/image" to="/data_throttled_image"/>
  	  <remap from="depth/image" to="/data_throttled_image_depth"/>
  	  <remap from="rgb/camera_info" to="/data_throttled_camera_info"/>
  	
      <param name="rgb/image_transport" type="string" value="compressed"/>
      <param name="depth/image_transport" type="string" value="compressedDepth"/>
	
	  <param name="queue_size" type="int" value="10"/>
	
	  <!-- RTAB-Map's parameters: do "rosrun rtabmap rtabmap (double-dash)params" to see the list of available parameters. -->
	  <param name="RGBD/PoseScanMatching" type="string" value="true"/>  <!-- Do odometry correction with consecutive laser scans -->
	  <param name="RGBD/LocalLoopDetectionSpace" type="string" value="true"/>  <!-- Local loop closure detection (using estimated position) with locations in WM -->
	  <param name="RGBD/LocalLoopDetectionTime" type="string" value="false"/>   <!-- Local loop closure detection with locations in STM -->
	  <param name="RGBD/OptimizeFromGraphEnd" type="string" value="true"/> 
	  <param name="LccIcp/Type" type="string" value="2"/>                      <!-- 0=No ICP, 1=ICP 3D, 2=ICP 2D -->
	  <param name="LccBow/MaxDepth" type="string" value="0.0"/>                <!-- 3D visual words maximum depth 0=infinity -->
	  <param name="LccBow/InlierDistance" type="string" value="0.1"/>          <!-- 3D visual words correspondence distance -->
      <param name="RGBD/OptimizeFromGraphEnd" type="string" value="false"/>    <!-- Optimize graph from initial node so /map -> /odom transform will be generated -->
    </node>
    
    <!-- Grid map assembler for rviz -->
    <node pkg="rtabmap" type="grid_map_assembler" name="grid_map_assembler">
    	<param name="unknown_space_filled" type="bool" value="false"/>
    </node>
  </group>
  
  <!-- Visualisation -->
      
  <node pkg="rviz" type="rviz" name="rviz" args="-d $(find rtabmap)/launch/config/demo_robot_mapping.rviz" output="screen"/>
  
  <node pkg="nodelet" type="nodelet" name="standalone_nodelet"  args="manager" output="screen"/>
   <node pkg="nodelet" type="nodelet" name="points_xyzrgb" args="load rtabmap/point_cloud_xyzrgb standalone_nodelet">
    <remap from="rgb/image"       to="data_throttled_image"/>
    <remap from="depth/image"     to="data_throttled_image_depth"/>
    <remap from="rgb/camera_info" to="data_throttled_camera_info"/>
    <remap from="cloud"           to="voxel_cloud" />
    
    <param name="rgb/image_transport" type="string" value="compressed"/>
    <param name="depth/image_transport" type="string" value="compressedDepth"/>
    
    <param name="queue_size" type="int" value="10"/>
    <param name="voxel_size" type="double" value="0.01"/>
  </node>

</launch>
